packages/trader/src/Modules/Trading/Components/Form/TradeParams/Duration/duration-wrapper.jsx

```diff
@@ -106,24 +97,20 @@ class DurationWrapper extends React.Component {
             onChange({ target: { name: 'expiry_type', value: 'duration' } });
         }
 
-        if (this.advancedHasWrongExpiry()) {
+        if (advancedHasWrongExpiry()) {
             onChange({ target: { name: 'expiry_type', value: advanced_expiry_type } });
         }
 
         if (duration !== current_duration) {
             onChangeUiStore({ name: `duration_${duration_unit}`, value: duration });
         }
 
-        this.assertDurationIsWithinBoundary(duration_unit, current_duration, onChangeUiStore, onChange);
-    }
+        assertDurationIsWithinBoundary(current_duration);
+    });
```

Comment:
No dependency? Do we need to call this effect always?

Reply:
Sorry you’re right I'll add

Reply:
I wanted to it add today, but I forgot :)